app/common/src/utilities/data/object.ts

```diff
@@ -9,6 +9,26 @@ export type Mutable<T> = {
   -readonly [K in keyof T]: T[K]
 }
 
+/** Whether two values are deeply equal. */
+export function deepEqual<T>(a: T, b: T) {
+  if (Object.is(a, b)) return true
+  if (typeof a !== typeof b) return false
+  if (typeof a !== 'object' || typeof b !== 'object') return Object.is(a, b)
```

Comment:
```suggestion
  if (Object.is(a, b)) return true
  if (typeof a !== typeof b) return false
  if (typeof a !== 'object' || typeof b !== 'object') return false
```
once we get to the 3rd `if`, we know that `Object.is(a, b) == false`. If it were `true`, we'd have returned from the first `if`.